README.md

```diff
@@ -1,2 +1,164 @@
+
+<!-- toc -->
+
+- [Running Container-to-Consul](#running-container-to-consul)
+  * [Using Docker](#using-docker)
+  * [Configuration](#configuration)
+    + [Consul](#consul)
+    + [Docker](#docker)
+    + [Logger](#logger)
+- [Container registration](#container-registration)
+  * [IP Address](#ip-address)
+  * [Service name](#service-name)
+  * [Tags](#tags)
+  * [Skipping registration](#skipping-registration)
+
+<!-- tocstop -->
+
+## Running Container-to-Consul
+
+### Using Docker
+
+```
+docker run -d \
+  -e c2c_consul__host=localhost \
+  -e c2c_consul__port=8500 \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  mekomsolutions/container2sul
+```
+
+### Configuration
+
+Container-to-Consul uses [rc](https://www.npmjs.com/package/rc) to import its configuration.
+
+The configuration can be overridden using any method supported by rc, for instance using a custom `.c2crc` file or by defining some environmental variables prefixed by `c2c_`.
+
+#### Consul
+
+Consul `host` and `port` can be configured:
+
+_.c2crc_
+```json
+{
+  "consul": {
+    "host": "localhost",
+    "port": 8500
+  },
+  "[..]": "[..]"
+}
+```
+
+#### Docker
+
+The `docker` section of the configuration is given to [dockerode](https://github.com/apocas/dockerode) constructor.
+
+Please refer to [dockerode](https://github.com/apocas/dockerode) documentation for supported options.
+
+Default configuration listens to Docker daemon on `/var/run/docker.sock` unix socket.
+
+_.c2crc_
+```json
+{
+  "[..]": "[..]",
+  "docker": {
+    "socketPath": "/var/run/docker.sock"
+  }
+}
+```
+
+#### Logger
+
+Container-to-Consul uses [bunyan](https://github.com/trentm/node-bunyan). The `logger` section of the configuraton object is passed to `bunyan.createLogger` method.
+
+Default configuration outputs logs at `info` level to `stdout`.
+
+_.c2crc_
+```json
+{
+  "[..]": "[..]",
+  "logger": {
+    "name": "container-to-consul"
+  }
+}
+```
+
+## Container registration
+
+Container-to-Consul relies on [Docker labels](https://docs.docker.com/engine/userguide/labels-custom-metadata/) to define the values sent to Consul.
+
+Internally, Container-to-Consul calls Consul low-level [catalog/register](https://www.consul.io/docs/agent/http/catalog.html#catalog_register) and [catalog/deregister](https://www.consul.io/docs/agent/http/catalog.html#catalog_deregister) API endpoints.
+
+Container registration is an opt-in process. By default, Container-to-Consul will not register any container in Consul. To register a container the `consul.register` label must be set to `true`.
+
+### IP Address
+
+By default, the registered ip address is the first one found in the `NetworkSettings` group from `docker inspect` result.
+
+If the container is running in `host` network mode, the ip address used is the `docker0` one.
+
+The registered ip address can be configured using a `consul.ip` label.
+
+```
+docker run --rm --name containertoConsul -ti -l consul.ip=10.0.0.1 alpine ash
```

Comment:
Let's have the documentation starting a container named `container-to-consul` if possible.